src/command-handlers/devices.ts

```diff
@@ -89,3 +94,41 @@ export async function removeAllDevices(devices: string[] | null, options: { all:
     }
     db.close();
 }
+
+export const registerNonInteractiveDevice = async (deviceName: string, options: { json: boolean }) => {
+    const {
+        secrets: { login },
+        db,
+    } = await connectAndPrepare({ autoSync: false });
+
+    const { type } = await get2FAStatusUnauthenticated({ login });
+
+    if (type === 'totp_login') {
+        throw new Error("You can't register a device when you have OTP at each login enabled.");
```

Comment:
Consider `You can't register a non interfactive device [...]` instead, so it's a bit more explicit.